newton/_src/solvers/kamino/_src/kinematics/joints.py

```
@@ -497,13 +497,14 @@ def _write_typed_joint_data(
         # Only write the constraint residual and velocity if the joint defines constraints
         # NOTE: This will be disabled for free joints
         if wp.static(num_cts > 0):
-            # Construct a 6D residual vector
-            j_theta_j = wp.static(get_joint_constraint_angular_residual_function(dof_type))(j_q_j)
-            j_p_j = wp.spatial_vectorf(*j_r_j, *j_theta_j)
-            # Store the joint constraint residuals
-            for j in range(num_cts):
-                r_j_out[cts_offset + j] = j_p_j[cts_axes[j]]
-                dr_j_out[cts_offset + j] = j_u_j[cts_axes[j]]
+            if cts_offset >= int32(0):
```

Ruff (0.16.1)

[error] Undefined name `int32`

(F821)